src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -472,8 +479,12 @@ namespace HVACUnitarySystem {
 				AirLoopControlInfo( AirLoopNum ).ReqstEconoLockoutWithCompressor = false;
 			}
 
-			if ( ( HeatActive ) && ( AirLoopControlInfo( AirLoopNum ).CanLockoutEconoWithCompressor || AirLoopControlInfo( AirLoopNum ).CanLockoutEconoWithHeating ) ) {
-				AirLoopControlInfo( AirLoopNum ).ReqstEconoLockoutWithHeating = true;
+			if ( present (HeatActive) ) {
+				if ( ( HeatActive ) && ( AirLoopControlInfo( AirLoopNum ).CanLockoutEconoWithCompressor || AirLoopControlInfo( AirLoopNum ).CanLockoutEconoWithHeating ) ) {
+					AirLoopControlInfo( AirLoopNum ).ReqstEconoLockoutWithHeating = true;
+				} else {
+					AirLoopControlInfo( AirLoopNum ).ReqstEconoLockoutWithHeating = false;
+				}
```

Comment:
result of zone, air loop, OA sys example file test. "HeatActive" flag wasn't passed in from MixedAir.